dev/tasks/java-jars/github.yml

```diff
@@ -22,8 +22,22 @@
 jobs:
 
   build-cpp-ubuntu:
-    name: Build C++ libraries Ubuntu
-    runs-on: ubuntu-latest
+    {% set arch = '${{ matrix.platform.arch }}' %}
+    name: Build C++ libraries Ubuntu {{ arch }}
+    runs-on: {{ '${{ matrix.platform.runs_on }}' }}
+    env:
+      # archery uses these environment variables
+      {% if arch == "x86_64" %}
```

Comment:
We can't use this here because `arch` is `'${{ matrix.platform.arch }}'`.
How about adding `archery_arch: amd64` and `archery_arch: arm64v8` to `platform` and use `ARCH: ${{ matrix.platform.archery_arch }}`?

Reply:
Thanks that makes sense. Done with [5fd2bec](https://github.com/apache/arrow/pull/36487/commits/5fd2bec4d23745bf87098ddc138e30ab2c480547)